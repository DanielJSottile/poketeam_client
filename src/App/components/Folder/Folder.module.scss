@import '../../../index.scss';

.button-styling {
  background-color: transparent;
  color: $super-light-cream;
  width: 100px;
  height: 75px;
  &:hover,
  &:hover:enabled {
    transform: translateY(-16px);
    transition: 0.3s;
    background-color: transparent;
    color: $azure;
    span {
      color: $super-light-cream;
    }
  }
}

.active {
  transform: translateY(-16px);
  transition: 0.3s;
  background-color: transparent;
  color: $azure;
  span {
    color: $super-light-cream;
  }
}

.font {
  position: relative;
  bottom: 100%;
  font-size: 0.9rem;
  font-family: $body-font;
  color: $med-dark-grey;
  padding: 0.3125rem;
}
