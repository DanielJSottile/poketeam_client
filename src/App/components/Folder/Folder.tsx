import React, { useContext } from 'react';
import Button from '../Button/Button';
import GeneralContext from '../../contexts/GeneralContext';

type Props = {
  /** id for folder*/
  id?: string;
  /** name for folder */
  folder_name?: string | undefined;
};

const Folder: React.FC<Props> = ({ id = '', folder_name = '' }) => {
  const { handleCurrentFolderClicked } = useContext(GeneralContext);

  return (
    <Button
      buttonClass="btn"
      id={id}
      onClickCallback={() => handleCurrentFolderClicked(folder_name, id)}
    >
      <i className="fas fa-folder"></i> {folder_name}
    </Button>
  );
};

export default Folder;
