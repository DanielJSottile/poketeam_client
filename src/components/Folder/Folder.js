import React, { Component } from 'react'

export default class Folder extends Component {

  handleOnClick = (e) => {
    e.preventDefault();
    console.log("hey you clicked me");
    const {folder} = e.target;

    // some kind of APIservice.GetTeams(folder.id)
  };

  render() {
    return (
      <button 
      id={this.props.id} 
      onClick={this.handleOnClick}>
        {this.props.folder_name}
      </button>
    );
  };
};
