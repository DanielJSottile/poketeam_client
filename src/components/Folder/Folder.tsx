import React, { useContext } from 'react';
import GeneralContext from '../../contexts/GeneralContext';

const Folder = (props: any) => {
  const GenCon = useContext(GeneralContext);
  const { handleCurrentFolderClicked } = GenCon;

  return (
    <button
      className="btn"
      id={props.id}
      onClick={() => handleCurrentFolderClicked(props.folder_name, props.id)}
    >
      <i className="fas fa-folder"></i> {props.folder_name}
    </button>
  );
};

export default Folder;
