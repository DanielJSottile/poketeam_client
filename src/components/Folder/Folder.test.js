import React from 'react';
import ReactDOM from 'react-dom';
//import { render } from '@testing-library/react';
import Folder from './Folder';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<Folder/>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Folder />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});