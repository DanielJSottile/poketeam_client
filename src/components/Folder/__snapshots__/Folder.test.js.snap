// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Folder/> renders without crashing 1`] = `
<button
  className="btn"
  onClick={[Function]}
>
  <i
    className="fas fa-folder"
  />
   
</button>
`;
